packages/flutter/test/widgets/autocomplete_test.dart

```diff
@@ -421,6 +421,130 @@ void main() {
+                fieldViewBuilder: (_, TextEditingController controller, FocusNode focusNode, __) {
+                  return TextField(controller: controller, focusNode: focusNode);
+                },
+                optionsViewBuilder: (_, __, ___) => const Placeholder())))));
+      await tester.showKeyboard(find.byType(TextField));
+      final Offset fieldTopLeft = tester.getTopLeft(find.byType(TextField));
+      final Offset optionsViewBottomLeft = tester.getBottomLeft(find.byType(Placeholder));
+      final Offset optionsViewCenter = tester.getCenter(find.byType(Placeholder));
+      expect(fieldTopLeft, offsetMoreOrLessEquals(optionsViewBottomLeft));
+      expect(optionsViewCenter.dy, lessThan(optionsViewBottomLeft.dy));
+    });
+
+    group('fieldViewBuilder not passed', () {
+      // Find the [SizedBox.shrink] that substitutes for a text field when
+      // [RawAutocomplete.fieldViewBuilder] is not passed.
+      final Finder sizedBoxShrinkFinder = find.descendant(
+        of: find.byType(RawAutocomplete<String>),
+        matching: find.byWidgetPredicate((Widget widget) {
+          return widget is SizedBox && widget.height == 0.0 && widget.width == 0.0;
+        }));
+
+      testWidgets('down', (WidgetTester tester) async {
+        final TextEditingController controller = TextEditingController();
+        final FocusNode focusNode = FocusNode();
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  TextField(controller: controller, focusNode: focusNode),
+                  RawAutocomplete<String>(
+                    key: GlobalKey(),
+                    textEditingController: controller,
+                    focusNode: focusNode,
+                    optionsViewOpenDirection: OptionsViewOpenDirection.down, // ignore: avoid_redundant_argument_values
+                    optionsBuilder: (_) => <String>['a'],
+                    optionsViewBuilder: (_, __, ___) => const Placeholder()),
+                ]))));
+        await tester.showKeyboard(find.byType(TextField));
+        final Offset sizedBoxBottomLeft = tester.getBottomLeft(sizedBoxShrinkFinder);
+        final Offset optionsViewTopLeft = tester.getTopLeft(find.byType(Placeholder));
+        final Offset optionsViewCenter = tester.getCenter(find.byType(Placeholder));
+        expect(sizedBoxBottomLeft, offsetMoreOrLessEquals(optionsViewTopLeft));
+        expect(optionsViewCenter.dy, greaterThan(optionsViewTopLeft.dy));
+      });
+
+      testWidgets('up', (WidgetTester tester) async {
+        final TextEditingController controller = TextEditingController();
+        final FocusNode focusNode = FocusNode();
+        await tester.pumpWidget(
+          MaterialApp(
+            home: Scaffold(
+              body: Column(
+                mainAxisAlignment: MainAxisAlignment.end,
```

Comment:
What's the reason for using `MainAxisAlignment.end` in this test case? Combined with swapping the order in the Column, it seems like this will put the RawAutocomplete at the bottom, same as it is in the other test case.

Reply:
The `RawAutocomplete` will be positioned *near* the bottom of the screen, but I expect the text field to still be below it.

The goal is a reasonable layout, at least as far as making sure (some of) the options view is onscreen.

In this test, where we ask the options view to render upwards above the text field, accepting the default `MainAxisAlignment.start` would mean letting the options view render entirely offscreen, above the top of the screen. We might have avoided that in a different way, like adding a [`Scaffold.appBar`](https://api.flutter.dev/flutter/material/Scaffold/appBar.html), but this way seemed fine.

Reply:
Ahh, I see — I was confusing the effect of this parameter with that of [Column.verticalDirection](https://api.flutter.dev/flutter/widgets/Flex/verticalDirection.html).

That reasoning makes sense. Rereading this test with the right understanding of that parameter in mind, I see that this and the other test are symmetrical just as one would hope.